tests/cnf/core/network/sriov/tests/qinq.go

```diff
@@ -434,6 +477,114 @@ var _ = Describe("QinQ", Ordered, Label(tsparams.LabelQinQTestCases), ContinueOn
+
+			By("Define and create sriov-network with 802.1ad S-VLAN")
+			vlan, err := strconv.Atoi(NetConfig.VLAN)
+			Expect(err).ToNot(HaveOccurred(), "Failed to convert VLAN value")
+			defineAndCreateSrIovNetworkWithQinQ(srIovNetworkDPDKDot1AD, srIovPolicyResNameVfioPci, dot1ad,
+				uint16(vlan))
+
+			By("Define and create sriov-network with 802.1q S-VLAN")
+			vlan, err = strconv.Atoi(NetConfig.VLAN)
+			Expect(err).ToNot(HaveOccurred(), "Failed to convert VLAN value")
+			defineAndCreateSrIovNetworkWithQinQ(srIovNetworkDPDKDot1Q, srIovPolicyResNameVfioPci, dot1q,
+				uint16(vlan))
+
+			By("Define and create a network attachment definition for dpdk container")
+			tapNad, err := define.TapNad(APIClient, nadCVLANDpdk, tsparams.TestNamespaceName, 0, 0, nil)
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Fail to define the Network-Attachment-Definition %s",
+				nadCVLANDpdk))
+			_, err = tapNad.Create()
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Fail to create Network-Attachment-Definition %s",
+				nadCVLANDpdk))
+		})
+
+		It("Verify network traffic over a 802.1ad QinQ tunnel between two DPDK pods on the same PF",
+			reportxml.ID("72636"), func() {
+				By("Define and create a container in promiscuous mode")
+				tcpDumpContainer := createPromiscuousClient(workerNodeList[0].Definition.Name,
+					tcpDumpNet1CMD)
+
+				By("Verify SR-IOV Device IDs for interface under test")
+				if sriovDeviceID != intelDeviceIDE810 {
+					Skip(fmt.Sprintf("The NIC %s does not support 802.1AD", sriovDeviceID))
+				}
+
+				By("Define and create a dpdk server container")
+				annotation := pod.StaticIPAnnotationWithMacAddress(srIovNetworkDPDKDot1AD, []string{}, dpdkServerMac)
+				testCmdServer := defineTestServerPmdCmd(dpdkClientMac, "${PCIDEVICE_OPENSHIFT_IO_SRIOVPOLICYVFIOPCI}")
+				fmt.Println(testCmdServer)
+				_ = defineAndCreateServerDPDKPod(serverNameDPDKDot1ad, workerNodeList[0].Definition.Name, annotation, testCmdServer)
+
+				By("Define and create a dpdk client container")
+				var annotationDpdk []*multus.NetworkSelectionElement
+				sVlan := pod.StaticIPAnnotationWithMacAddress(srIovNetworkDPDKDot1AD, []string{}, dpdkClientMac)
+				cVlan := pod.StaticAnnotation(nadCVLANDpdk)
+				annotationDpdk = append(annotationDpdk, sVlan[0], cVlan)
+
+				clientDpdk := defineAndCreateClientDPDKPod(clientNameDPDKDot1ad, workerNodeList[0].Definition.Name, annotationDpdk)
+
+				testCmdClient := defineTestClientPmdCmd(dpdkClientMac, "${PCIDEVICE_OPENSHIFT_IO_SRIOVPOLICYVFIOPCI}")
+				fmt.Println(testCmdClient)
+				By("Validate dpdk_testpmd traffic from the server to the client using CVLAN100.")
+				rxTrafficOnClientPod(clientDpdk, testCmdClient[0])
+
+				By("Validate that the TCP traffic is double tagged")
+				readAndValidateTCPDump(tcpDumpContainer, tcpDumpReadFileCMD, tcpDumpDot1ADOutput)
+			})
+
+		It("Verify network traffic over a 802.1q QinQ tunnel between two DPDK pods on the same PF",
```

Comment:
Only diff between 2 dpdk tests is srIovNetworkDPDKDot1Q and srIovNetworkDPDKDot1AD. Could you create one common func and add it as an atribute

Reply:
lets talk about it first

Reply:
?

Reply:
When are going to discuss it?